luxonis_train/core/archiver.py

```diff
@@ -0,0 +1,396 @@
+            )
+
+    def _get_onnx_outputs(self, executable_path):
+        """Get outputs of an ONNX model executable.
+
+        @type executable_path: str
+        @param executable_path: Path to model executable file.
+        """
+
+        outputs_dict = {}
+        model = onnx.load(executable_path)
+        for output in model.graph.output:
+            tensor_type = output.type.tensor_type
+            dtype_idx = tensor_type.elem_type
+            dtype = str(onnx.helper.tensor_dtype_to_np_dtype(dtype_idx))
+            outputs_dict[output.name] = {"dtype": dtype}
+        return outputs_dict
+
+    def _add_output(self, name: str, dtype: str) -> None:
+        """Add output to self.outputs.
+
+        @type name: str
+        @param name: Name of the output layer.
+        @type dtype: str
+        @param dtype: Data type of the output data (e.g., 'float32').
+        """
+
+        self.outputs.append({"name": name, "dtype": dtype})
+
+    def _get_classes(self, head_family):
+        if head_family.startswith("Classification"):
+            return self.dataset_metadata._classes["class"]
+        elif head_family.startswith("Object"):
+            return self.dataset_metadata._classes["boxes"]
+        elif head_family.startswith("Segmentation"):
+            return self.dataset_metadata._classes["segmentation"]
+        elif head_family.startswith("Keypoint"):
+            return self.dataset_metadata._classes["keypoints"]
+        else:
+            raise ValueError(
+                f"No classes found for the specified head family ({head_family})"
+            )
+
+    def _get_head_specific_parameters(
+        self, head_name, head_alias, executable_path
+    ) -> dict:
+        """Get parameters specific to head.
+
+        @type head_name: str
+        @param head_name: Name of the head (e.g. 'EfficientBBoxHead').
+        @type head_alias: str
+        @param head_alias: Alias of the head (e.g. 'detection_head').
+        @type executable_path: str
+        @param executable_path: Path to model executable file.
+        """
+
+        parameters = {}
+        if head_name == "ClassificationHead":
+            parameters["is_softmax"] = self._is_softmax(executable_path)  # TODO: test
+        elif head_name == "EfficientBBoxHead":
+            parameters["subtype"] = "yolov6"
+            head_node = self.lightning_module._modules["nodes"][head_alias]
+            parameters["iou_threshold"] = head_node.iou_thres
+            parameters["conf_threshold"] = head_node.conf_thres
+            parameters["max_det"] = head_node.max_det
+            # head_outputs["n_keypoints"] # TODO: implement
+            # head_outputs["n_prototypes"] # TODO: implement
+            # head_outputs["prototype_output_name"] # TODO: implement
+        elif head_name == "ObjectDetectionSSD":
+            raise NotImplementedError
+            # head_outputs["anchors"] # TODO: implement
```

Comment:
I don't think we support any architectures that will require anchors at the moment, just FYI

Reply:
`ImplicitKeypointBBoxHead` head does require anchors.

Reply:
Ah thanks for pointing that out. Actually need to make a change to NN archive then

Reply:
(But realizing I need all the object detection parameters in the keypoint head anyways in NN archive)